File: OpenCV/05_color_histogramas/cpp/main.cpp
#include <opencv2/opencv.hpp>
int main(){ auto img=cv::imread("input.jpg"); if(img.empty()) return 1; cv::Mat hsv,g,eq,cl; cv::cvtColor(img,hsv,cv::COLOR_BGR2HSV); cv::cvtColor(img,g,cv::COLOR_BGR2GRAY); cv::equalizeHist(g,eq); cv::createCLAHE(2.0,{8,8})->apply(g,cl); cv::imwrite("output_equalized.png",eq); cv::imwrite("output_clahe.png",cl); }

File: OpenCV/05_color_histogramas/python/main.py
import cv2
img=cv2.imread("input.jpg"); assert img is not None
hsv=cv2.cvtColor(img,cv2.COLOR_BGR2HSV); gray=cv2.cvtColor(img,cv2.COLOR_BGR2GRAY); eq=cv2.equalizeHist(gray); clahe=cv2.createCLAHE(2.0,(8,8)).apply(gray)
cv2.imwrite("output_hsv_h.png",hsv[:,:,0]); cv2.imwrite("output_equalized.png",eq); cv2.imwrite("output_clahe.png",clahe)
